### DISCUSSION TITLE
Searching for equivalent to Boto3's `RDS.Client.generate_db_auth_token`

### DISCUSSION THREAD

**Discussion by robjtede** [awslabs/aws-sdk-rust#1049] [Q&A] [answered]
We're wanting to use IRSA authentication for more resources. We've successfully done so for S3 using `aws-sdk-s3` but we're now looking at databases which is a more complex interaction because `aws-sdk-rds` is not the DB client per-se.

Section 9 of [this guide](https://aws.amazon.com/blogs/containers/using-iam-database-authentication-with-workloads-running-on-amazon-eks/) shows use of Boto3's [`generate_db_auth_token`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds/client/generate_db_auth_token.html#RDS.Client.generate_db_auth_token) method to just obtain a password which can be used with any DB client.

I can't see any equivalent in `aws-sdk-rds` unless I've missed a more fundamental concept as to the nature of acceptable auth-tokens-as-passwords and this is already possible via a different crate.

**Comment by ysaito1001** [comment, marked as answer]
Hi @robjtede, thank you for asking the question on `generate_db_auth_token`. We acknowledge that similar requests have been asked in the past (you can see a workaround in [this issue](https://github.com/awslabs/aws-sdk-rust/issues/951), for instance).

The workaround in the above issue requires additional code; The reason `generate_db_auth_token` is not part of the `aws-sdk-rds` crate is that the function is not a Smithy-modeled operation, but a rather library function that needs to be hand-written (as opposed to code-generated by smithy-rs). For such a feature, you will see the [high-level-library](https://github.com/awslabs/aws-sdk-rust/labels/high-level-library) label within this repository.

Essentially, if labeled as `high-level-library`, it means we need to figure out how to house those high-level libraries, separately from code-generated SDKs.
